keytree/src/lib.rs

```diff
@@ -50,6 +50,39 @@ impl Xprv {
             precompressed_pubkey: self.precompressed_pubkey,
         }
     }
+
+    /// Serializes this Xprv to a sequence of bytes.
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut buf = [0u8; 64];
+        buf[..32].copy_from_slice(&self.scalar.to_bytes());
+        buf[32..].copy_from_slice(&self.dk);
+        buf
+    }
+
+    /// Decodes an Xprv from a 64-byte array, and fails if the provided array is not
+    /// exactly 64 bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 64 {
+            return None;
+        }
+
+        let (scslice, dkslice) = bytes.split_at(32);
+        let mut scalar_bytes = [0u8; 32];
+        scalar_bytes.copy_from_slice(&scslice[..]);
```

Comment:
`scslice` is already a slice, so it can be used directly w/o extra reslicing (it's cheap, but still needless operation):

```suggestion
        scalar_bytes.copy_from_slice(scslice);
```

however, see the above comment - i suggest slicing the `bytes` in place instead of `split_at`.